### DISCUSSION TITLE
Can't install ocp

### DISCUSSION THREAD

**Discussion by kTag** [Homebrew/discussions#23] [Everyday usage]
`% brew install ocp  
Updating Homebrew...  
==> Downloading https://github.com/mywave82/opencubicplayer/commit/04368dc54e649050517fe3a058e919fb5fb5f150.diff?full_index=1  
Already downloaded:   /Users/admin/Library/Caches/Homebrew/downloads/2c7a4e7a37c83fe67d552a56ea659512be2fee170a7614898d29002637185b27--04368dc54e649050517fe3a058e919fb5fb5f150.diff  
==> Downloading https://github.com/mywave82/opencubicplayer/commit/1907b8f85e3e5fa61a02ad0ca3ce9bd30bfc0ea6.diff?full_index=1  
Already downloaded:   /Users/admin/Library/Caches/Homebrew/downloads/a2ffd9c1b34026d64d3452fb5ad03d596cffe5bda873b2b4edfaf4aba6ba7aaa--1907b8f85e3e5fa61a02ad0ca3ce9bd30bfc0ea6.diff  
==> Downloading https://stian.cubic.org/ocp/ocp-0.2.1.tar.bz2  
Already downloaded:   /Users/admin/Library/Caches/Homebrew/downloads/b532c7dc6e9783469da29b6010f3d4086be38aa3a38c876ddacd0ed00ea8fa42--ocp-0.2.1.tar.bz2  
==> Patching  
==> Applying 04368dc54e649050517fe3a058e919fb5fb5f150.diff  
patching file playgmi/timidityplay.c
==> Applying 1907b8f85e3e5fa61a02ad0ca3ce9bd30bfc0ea6.diff
patching file playhvl/hvlplay.h
==> ./configure --prefix=/Users/admin/Applications/homebrew/Cellar/ocp/0.2.1 --without-x11 --without-sdl --without-sdl2 --without-desktop_file_install
Last 15 lines from /Users/admin/Library/Logs/Homebrew/ocp/01.configure:
checking whether make sets $(MAKE)... yes
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking whether byte ordering is bigendian... no
checking gcc version... configure: error: 12.0.0, bad`

What can I do?

**Comment by gromgit** [comment]
Run `brew gist-logs` and post the Gist URL that it generates. The build output you posted:
```
...
checking gcc version... configure: error: 12.0.0, bad
```
is incomplete.

**Reply by kTag** [reply]
`% brew gist-logs ocp
Error: A newer Command Line Tools release is available.
Update them from Software Update in System Preferences or run:
  softwareupdate --all --install --force`

Then running the update I get:
```
Software Update Tool

Finding available software
No updates are available.
```

What's going on?

**Reply by Rylan12** [reply]
Yeah, the CLT update process is a pain. It may be worth manually checking in System Preferences to see if there are any updates.

Otherwise, here are two things you can try:

1. Remove the existing CLT (you can find the location by running `xcode-select -p`) and reinstall using `xcode-select --install`
1. Go to the [Apple Developer downloads](https://developer.apple.com/download/more/) page and manually download and install the latest (stable) version of the CLT (as I'm writing this it looked like the latest version is "Command Line Tools for Xcode 12").

I've not experienced the specific error you've encountered but I do have issues and need to reinstall the CLT every once in a while. I usually try to use option 1 and, when it fails, end up using option 2.

**Reply by kTag** [reply]
I went directly with option 2, worked like a charm:
https://gist.github.com/62ce24186ff33ec028a50a276171cd0e

**Comment by Rylan12** [comment]
From the logs:

```
Warning: Your Homebrew's prefix is not /usr/local.
Some of Homebrew's bottles (binary packages) can only be used with the default
prefix (/usr/local).
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Discourse, Twitter or IRC. You are responsible for resolving any issues you
experience while you are running this unsupported configuration.
```
